src/shared/pages/team/editTeam/editTeam.tsx

```diff
@@ -80,20 +136,30 @@ const EditTeam: React.FC<EditTeamProps> = ({ csrfToken, history, match }) => {
                 <div className="govuk-grid-column-two-thirds-from-desktop">
                     <form method="POST" onSubmit={handleSubmit}>
                         <input type="hidden" name="_csrf" value={csrfToken} />
-                        <Text
+                        {(hasRole('RENAME_TEAM')) && <Text
                             label="New team name"
                             name="newTeamName"
                             type="text"
                             updateState={
                                 ({ value }) => dispatch({ type: 'SetNewTeamName', payload: value as string })
                             }
                             value={state.newDisplayName}
-                        />
+                        />}
+                        {(hasRole('REASSIGN_TEAM_UNIT')) && <TypeAhead
+                            clearable={true}
+                            disabled={false}
+                            getOptions={getUnitsForTypeAhead}
+                            label={'Unit'}
+                            name={'unit'}
+                            value={{ label: state.unit?.label || '', value: '' }}
 
+                            onSelectedItemChange={
+                                (item  ) => dispatch({ type: 'SetUnit', payload: item })
+                            }
+                        />}
                         <div className="govuk-form-group">
                             <button type="submit" className="govuk-button">Update</button>
```

Comment:
If the user has none of the above roles, what do they see on this page? Would it be a blank form with a submit button?

Reply:
If the user has neither role, the link in the TeamView should never appear to allow them to get to this page.

Reply:
Ok, is this route role protected so a user can't access the page by using the URL directly?